app/scripts/background.js

```diff
@@ -721,10 +724,28 @@ function setupController(initState, initLangCode) {
         ),
       );
 
-    // Finally, reject all approvals managed by the ApprovalController
+    // Finally, resolve snap dialog approvals on Flask and reject all the others managed by the ApprovalController.
+
+    ///: BEGIN:ONLY_INCLUDE_IN(flask)
+    Object.values(controller.approvalController.state.pendingApprovals).forEach(
+      ({ id, type }) => {
+        if (type.startsWith(RestrictedMethods.snap_dialog)) {
+          controller.approvalController.accept(id, null);
```

Comment:
This doesn't mirror the results of all `snap_dialog` types. Confirmations either return `true` or `false`.